src/confload.nim

```diff
@@ -177,14 +177,11 @@ proc loadAllConfigs*() =
   let
     toStream = newStringStream
     stack    = newConfigStack()
+    exeName  = getMyAppPath().splitPath().tail
 
-  case getMyAppPath().splitPath().tail
+  case exeName
   of "docker":
-    if "docker" notin params:
-      if len(params) != 0 and params[0] == "chalk":
-        params = params[1 .. ^1]
```

Comment:
@viega what was this guarding against? I cant think of a case where this can apply. if your executable name is `docker` when would `chalk` be the first parameter?

Reply:
Indirect recursion due to multiple chalks lying around.

Reply:
did a bunch of experiments locally and dont think recursion applies here, at least not anymore

there are a couple of cases where recursion can happen:

* `chalk docker ...` or `cp chalk docker && docker ...` (which is chalk internally). in this case the actual `docker` to be executed is found via `findExePath` which will found exe on `PATH` but will ignore all other chalks so recursion should not be possible there
  https://github.com/crashappsec/chalk/blob/b8b3481d53374d5bc5d251594b47c4708d54ebd7/src/util.nim#L355
  verified with a nested recursion with (excuse some of the fish shell conventions):
  ```
  ➜ PATH=$PWD/one:$PWD/two:$PWD/three:(echo $PATH | tr ' ' ':') ./docker version
  ```
  that correctly avoided all chalk dockers in `one`/`two`/`three` although noticed subscan was used on all dockers on PATH even after non-chalk docker was found. as subscan is kind of expensive addressed that in [`b8b3481` (#257)](https://github.com/crashappsec/chalk/pull/257/commits/b8b3481d53374d5bc5d251594b47c4708d54ebd7)
* mac codec does strange things due to a bash script wrapper for chalk. in that case that also works as expected as underlying binary name stays unchanged.
  https://github.com/crashappsec/chalk/blob/b8b3481d53374d5bc5d251594b47c4708d54ebd7/src/plugins/codecMacOs.nim#L18
  https://github.com/crashappsec/chalk/blob/b8b3481d53374d5bc5d251594b47c4708d54ebd7/src/plugins/codecMacOs.nim#L30
  so if you `cp chalk docker && ./docker` the underlying executable will still be called `docker` from chalk perspective and so same logic as above applies
* for all other cases recursion should be picked up specifically by recursion check in chalk:
  https://github.com/crashappsec/chalk/blob/b8b3481d53374d5bc5d251594b47c4708d54ebd7/src/norecurse.nim#L16-L38
* final case I can think of is `chalk exec -- chalk exec -- echo foo` but I dont think we ever addressed what we actually want to do there. do we run ignore inner chalk there or skip it and execute inner command. there are arguments in both directions. but either way dont think this case applies here as the exe name wont be `docker` there and the if statement here was only protection `./docker ...`

I currently cant compile on intel mac so cant verify 100% its working legit on mac but I did some POCs with similar bash scripts and can verify exe name wasnt changed by the mac codec

should we keep it simple here and if it causes any issues we can address later with a comment explaining the rational for the fix?